module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -59,92 +59,94 @@ where
         + TxHooks<Context = C>
         + ApplyBlobHooks<Context = C, BlobResult = SequencerOutcome>,
 {
-    fn init_sequencer_and_get_working_set(
+    pub fn new(storage: C::Storage, runtime: RT) -> Self {
+        Self {
+            runtime,
+            current_storage: storage,
+            checkpoint: None,
+            phantom_vm: PhantomData,
+        }
+    }
+
+    pub(crate) fn apply_batch(
         &mut self,
         sequencer: &[u8],
-        batch_data_and_hash: &BatchDataAndHash,
-    ) -> Result<WorkingSet<C::Storage>> {
+        batch: impl Buf,
+    ) -> ApplyBatchResult<BatchReceipt<SequencerOutcome, TxEffect>> {
+        let batch_data_and_hash = BatchDataAndHash::new::<C>(batch);
         debug!(
             "Applying batch from sequencer: 0x{}",
             hex::encode(sequencer)
         );
+
+        // Point A:
+        // Initial batch workspace
         let mut batch_workspace = self
             .checkpoint
             .take()
             .expect("Working_set was initialized in begin_slot")
             .to_revertable();
 
+        // ApplyBlobHook: begin
         if let Err(e) =
             self.runtime
                 .begin_blob_hook(sequencer, &batch_data_and_hash.data, &mut batch_workspace)
         {
             error!(
-                "Error: The transaction was rejected by the 'enter_apply_blob' hook. Skipping batch without slashing the sequencer: {}",
+                "Error: The batch was rejected by the 'begin_blob_hook' hook. Skipping batch without slashing the sequencer: {}",
                 e
             );
+            // TODO: Should it commit?
             self.checkpoint = Some(batch_workspace.revert());
             return Err(ApplyBatchError::Ignored(batch_data_and_hash.hash));
         }
+        // Point B: sequencer has paid
```

Comment:
nit: It's a little unclear why `Sequencer has paid` is a `point`, but `ApplyBlobHook: begin` isn't. What do the `Point`s correspond to?

Reply:
Points comment are actually leftovers from drafting and discussion,